osquery/tables/system/windows/windows_eventlog.cpp

```diff
@@ -0,0 +1,237 @@
+/**
+ *  Copyright (c) 2014-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed in accordance with the terms specified in
+ *  the LICENSE file found in the root directory of this source tree.
+ */
+
+#include <Windows.h>
+#include <winevt.h>
+
+#include <boost/algorithm/string.hpp>
+#include <boost/foreach.hpp>
+#include <boost/property_tree/json_parser.hpp>
+#include <boost/property_tree/ptree.hpp>
+#include <boost/property_tree/xml_parser.hpp>
+
+#include <osquery/core/tables.h>
+#include <osquery/logger/logger.h>
+
+#include <osquery/core/windows/wmi.h>
+#include <osquery/events/windows/windowseventlogparser.h>
+#include <osquery/sql/dynamic_table_row.h>
+#include <osquery/utils/conversions/join.h>
+#include <osquery/utils/conversions/split.h>
+#include <osquery/utils/conversions/windows/strings.h>
+
+namespace pt = boost::property_tree;
+
+namespace osquery {
+namespace tables {
+
+const std::string kEventLogXmlPrefix = "<QueryList><Query Id=\"0\">";
+const std::string kEventLogXmlSuffix = "</Query></QueryList>";
+
+const int kNumEventsBlock = 1024;
+
+void parseWelXml(QueryContext& context,
+                 std::wstring& xml_event,
+                 RowYield& yield) {
+  auto row = make_table_row();
+  pt::ptree propTree;
+  WELEvent windows_event;
+  auto status = parseWindowsEventLogXML(propTree, xml_event);
+  status = parseWindowsEventLogPTree(windows_event, propTree);
+
+  row["time"] = INTEGER(windows_event.osquery_time);
+  row["datetime"] = SQL_TEXT(windows_event.datetime);
+  row["channel"] = SQL_TEXT(windows_event.source);
+  row["provider_name"] = SQL_TEXT(windows_event.provider_name);
+  row["provider_guid"] = SQL_TEXT(windows_event.provider_guid);
+  row["eventid"] = INTEGER(windows_event.event_id);
+  row["task"] = INTEGER(windows_event.task_id);
+  row["level"] = INTEGER(windows_event.level);
+  row["pid"] = INTEGER(windows_event.pid);
+  row["tid"] = INTEGER(windows_event.tid);
+
+  row["keywords"] = SQL_TEXT(windows_event.keywords);
+  row["data"] = SQL_TEXT(windows_event.data);
+
+  if (context.hasConstraint("time_range", EQUALS)) {
+    auto time_range = context.constraints["time_range"].getAll(EQUALS);
+    row["time_range"] = SQL_TEXT(*time_range.begin());
+  } else {
+    row["time_range"] = SQL_TEXT("");
+  }
+
+  if (context.hasConstraint("timestamp", EQUALS)) {
+    auto timestamp = context.constraints["timestamp"].getAll(EQUALS);
+    row["timestamp"] = SQL_TEXT(*timestamp.begin());
+  }
+
+  if (context.hasConstraint("xpath", EQUALS)) {
+    auto xpaths = context.constraints["xpath"].getAll(EQUALS);
+    row["xpath"] = SQL_TEXT(*xpaths.begin());
+  }
+
+  yield(std::move(row));
+}
+
+void parseQueryResults(QueryContext& context,
```

Comment:
As a nit. this isn't parsing query results, it's iterating over a handle to a stream.